web_client/src/components/VtkViewer.vue

```diff
@@ -276,121 +155,258 @@ export default {
+          (manipulator) => manipulator.getClassName() === 'vtkMouseRangeManipulator',
+        );
+      if (targetManipulator) {
+        targetManipulator.setDragEnabled(false);
       }
-    },
+      setTimeout(() => {
+        resized.value = true;
+      });
+    }
+    function initializeCamera() {
+      const camera = props.view.getCamera();
+      const orientation = representation.value.getInputDataSet().getDirection();
+
+      let newViewUp = VIEW_ORIENTATIONS[renderOrientation.value][name.value].viewUp.slice();
+      let newDirectionOfProjection = VIEW_ORIENTATIONS[
+        renderOrientation.value
+      ][name.value].directionOfProjection;
+      newViewUp = findClosestColumnToVector(
+        newViewUp,
+        orientation,
+      );
+      newDirectionOfProjection = findClosestColumnToVector(
+        newDirectionOfProjection,
+        orientation,
+      );
+
+      camera.setDirectionOfProjection(...newDirectionOfProjection);
+      camera.setViewUp(...newViewUp);
+
+      props.view.resetCamera();
+      fill2DView(props.view);
+    }
+    function prepareViewer() {
+      initializeView();
+      initializeSlice();
+      initializeCamera();
+      updateCrosshairs();
+      renderSubscription.value = props.view.getInteractor().onRenderEvent(() => {
+        updateCrosshairs();
+      });
+      resizeObserver.value = new window.ResizeObserver((entries) => {
+        if (entries.length === 1 && viewer.value && crosshairsCanvas.value) {
+          const width = viewer.value.clientWidth;
+          const height = viewer.value.clientHeight;
+          crosshairsCanvas.value.width = width;
+          crosshairsCanvas.value.height = height;
+          crosshairsCanvas.value.style.width = `${width}px`;
+          crosshairsCanvas.value.style.height = `${height}px`;
+          initializeCamera();
+          updateCrosshairs();
+        }
+      });
+      resizeObserver.value.observe(viewer.value);
+      applyCurrentWindowLevel();
+    }
+    function keyPress(event) {
+      if (['TEXTAREA', 'INPUT'].includes(document.activeElement.tagName)) return;
+      switch (event.key) {
+        case keyboardBindings.value[0]:
+          changeSlice(slice.value - 1);
+          break;
+        case keyboardBindings.value[1]:
+          changeSlice(slice.value + 1);
+          break;
+        case keyboardBindings.value[2]:
+          toggleFullscreen();
+          break;
+        default:
+          break;
+      }
+    }
+
+    watch(slice, (newSlice) => {
+      representation.value.setSlice(newSlice);
+      if (setCurrentVtkIndexSlices) {
+        setCurrentVtkIndexSlices({
+          indexAxis: ijkMapping[trueAxis(name.value)],
+          value: representation.value.getSliceIndex(),
+        });
+      }
+    });
+    watch(iIndexSlice, updateCrosshairs);
+    watch(jIndexSlice, updateCrosshairs);
+    watch(kIndexSlice, updateCrosshairs);
+    watch(showCrosshairs, updateCrosshairs);
+    watch(representation, () => {
```

Comment:
I noticed you moved away from using view to representation. This makes sense to me - as representation is more limited in scope than the view and we are using representation throughout the rest of the component. Was that the reason you moved to representation? Want to make sure I'm following!

Reply:
Yes, that's half the reason. The other half is that since the composition api has much clearer typing, it actually threw an error when trying to use `props.view` as the target for a `watch` clause, since that object is a VTK object, not a vue object. When it had been written as `this.view`, it didn't really know what type it was, so it didn't complain, but it caused the `watch` clause to fire unreliably when unrelated changes were made to the VTK object.